### ISSUE TITLE
What is PrivateConversationState? 

### ISSUE THREAD

**Issue by cleemullins** [microsoft/botframework-sdk#5488]
The C# SDK has a class called [PrivateConversationState](https://github.com/microsoft/botbuilder-dotnet/blob/8ff8744437c9006910228ae01185f40117f41a57/libraries/Microsoft.Bot.Builder/PrivateConversationState.cs).

The JS SDK does not. 

The Python SDK does not. 

Do we:
1. Remove this from C#
2. Add this to JS + Python?

It's unclear from context the intent or scope of this code.

**Comment by EricDahlvang** [comment]
This does exist in the js sdk: https://github.com/microsoft/botbuilder-js/blob/master/libraries/botbuilder-core/src/privateConversationState.ts

Docs describing Private Conversation State can be found here: https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-concept-state?view=azure-bot-service-4.0#state-management

**Comment by v-kydela** [comment]
The documentation is most likely sufficient to explain what private conversation state is, but I can paraphrase and elaborate if it's helpful.

The three bot state "buckets" are user state, conversation state, and private conversation state. When a bot accesses its bot state on a turn, it uses a key derived from the incoming activity. User state uses the user ID from the `from` property as its key. Conversation state uses the conversation ID as its key. Private conversation state uses a combination of the user ID and the conversation ID. You can see in the code you linked how these properties are combined:

```c#
return $"{channelId}/conversations/{conversationId}/users/{userId}";
```

If a channel only allows one conversation between a user and a bot with no group conversations or anything then the user ID and conversation ID effectively serve the same purpose and it doesn't matter which bucket you use. But in channels that allow multiple conversations with each user like the email channel, private conversation state can be used to differentiate between different conversations with the same user and keep the state from those conversations separate, as opposed to user state which would get reused between conversations. Alternatively, if a channel allows group conversations then a bot might get activities with the same conversation ID from different users, and so private conversation can allow the bot to keep track of state for each user in the conversation whereas conversation state would be shared between all of the users in the conversation.

**Comment by Stevenic** [comment]
For a bit more context... In v3 we stored the current conversations dialog stack in `PrivateConversationState` because we assumed the bot was always talking to a single user and this would allow it to track the conversation it was having with said user.  This quickly became problematic when channels like skype rolled out support for group conversations.  The issue was that one user in the group would start a conversation with the bot and other users might try to answer the bot but it would ignore them because it was tracking every conversation with each user separately. We ended up adding an option to switch to using `ConversationState` to track dialogs at the group level but we had to the leave the default as is for v3 because of compat issues.

In v4 I switched to using `ConversationState` by default as it maintains one state bag per group but I left `PrivateConversationState` in the SDK because there are still cases where it may be of use. Tracking individual answers to a poll question for instance. But even in that scenario it's probably just as easy to do it a group level using a map of responses keyed of userId.
